packages/ui/src/components/inputs/text-input.tsx

```diff
@@ -25,19 +25,23 @@ const TextInput = forwardRef<HTMLInputElement, TextInputProps>((props, ref) => {
   return (
     <ControlGroup className={wrapperClassName}>
       {!!label && (
-        <Label disabled={props.disabled} optional={optional} htmlFor={inputId}>
+        <Label disabled={restProps.disabled} optional={optional} htmlFor={inputId}>
           {label}
         </Label>
       )}
 
       <BaseInput {...restProps} ref={ref} theme={theme} id={inputId} />
 
       {error ? (
-        <FormCaption theme="danger">{error}</FormCaption>
+        <FormCaption disabled={restProps.disabled} theme="danger">
+          {error}
+        </FormCaption>
       ) : warning ? (
-        <FormCaption theme="warning">{warning}</FormCaption>
+        <FormCaption disabled={restProps.disabled} theme="warning">
+          {warning}
+        </FormCaption>
       ) : caption ? (
-        <FormCaption>{caption}</FormCaption>
+        <FormCaption disabled={restProps.disabled}>{caption}</FormCaption>
```

Comment:
can we de-structure `disabled` at the top  and pass it simply as `disabled` rather repeating as `restProps.disabled` in multiple places?

Reply:
Fixed